android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1272,51 +1273,35 @@ public void showSearch()
   @Override
   public void updateMenu()
   {
-    boolean isVisible = adjustMenuLineFrameVisibility();
-    if (!isVisible)
-      return;
+    final RoutingController controller = RoutingController.get();
 
-    if (RoutingController.get().isNavigating())
+    if (controller.isNavigating())
     {
       mNavigationController.show(true);
       closeSearchToolbar(false, false);
       mMainMenu.setState(MainMenu.State.NAVIGATION, isFullscreen());
       return;
     }
 
-    if (RoutingController.get().isPlanning())
-    {
-      mMainMenu.setState(MainMenu.State.ROUTE_PREPARE, isFullscreen());
-      return;
-    }
-
-    mMainMenu.setState(MainMenu.State.MENU, isFullscreen());
-  }
-
-  private boolean adjustMenuLineFrameVisibility()
```

Comment:
This method is used only by `public void updateMenu()`. So I moved it's body into `updateMenu()` method.